zml/tensor.zig

```diff
@@ -3629,6 +3629,27 @@ pub const Tensor = struct {
             }
         }.binaryOpHelper;
     }
+
+    /// Insert code that will print the content of the given buffer at runtime.
+    /// Only for debug purpose, it has the following limitations:
+    /// * only supported on Cuda atm
+    /// * only prints the first 1024 values
+    /// * pre allocates a buffer on the host to copy the content of the device buffer,
+    /// this buffer won't be freed. You will have one buffer per "print" call in the IR.
+    /// * does device to host synchronization so it will slow down the program execution.
+    pub fn print(input: Tensor) Tensor {
+        return ops.addHostCallback(&printCallback, input);
```

Comment:
Maybe we should log a warning or something when not running on Cuda ?

Reply:
I mean RTFM :-)